x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/SecurityActionMapper.java

```diff
@@ -41,6 +44,13 @@ public static String action(String action, TransportRequest request) {
                     return CLUSTER_PERMISSION_ANALYZE;
                 }
             }
+            case CLUSTER_PERMISSION_PAINLESS_EXECUTE -> {
+                assert request instanceof SingleShardRequest;
+                String index = ((SingleShardRequest<?>) request).index();
+                if (index != null) {
+                    return INDICES_PERMISSION_PAINLESS_EXECUTE;
```

Comment:
while it is pretty clear that we want to make this case an indices read action, and the request is already an indices request (it will not contain wildcards hence there is no need for it to implement Replaceable), I wonder if we are still ok with the action without the index being a cluster admin level action.

Reply:
I think we're good like this.
If we made a no-index action an index privilege, then it would have to fail because we won't know how to authorize it (which index would we check?)
Leaving it as a cluster privilege will at least give it a chance to succeed.